test/integration/scheduler/predicates_test.go

```diff
@@ -1106,7 +1108,95 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				return nil
 			},
 		},
-		// TODO(#91111): Add more test cases.
+		{
+			name: "other pod gets added",
+			init: func(cs kubernetes.Interface, _ string) error {
+				node, err := createNode(cs, "node-1", nil)
+				if err != nil {
+					return fmt.Errorf("cannot create node: %v", err)
+				}
+				// TODO: we could directly initialize the node with labels
+				if err := utils.AddLabelsToNode(cs, node.Name, map[string]string{"region": "test"}); err != nil {
+					return fmt.Errorf("cannot add labels to node: %v", err)
+				}
+				return nil
+			},
+			pod: &pausePodConfig{
+				Name: "pod-1",
+				Affinity: &v1.Affinity{
+					PodAffinity: &v1.PodAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"pod-with-affinity": "true",
+									},
+								},
+								TopologyKey: "region",
+							},
+						},
+					},
+				},
+			},
+			update: func(cs kubernetes.Interface, ns string) error {
+				podConfig := &pausePodConfig{
+					Name:      "pod-with-affinity",
+					Namespace: ns,
+					Labels: map[string]string{
+						"pod-with-affinity": "true",
+					},
+				}
+				if _, err := createPausePod(cs, initPausePod(podConfig)); err != nil {
+					return fmt.Errorf("cannot create pod: %v", err)
+				}
+				return nil
+			},
+		},
+		{
+			name: "scheduled pod gets updated",
+			init: func(cs kubernetes.Interface, ns string) error {
+				node, err := createNode(cs, "node-1", nil)
+				if err != nil {
+					return fmt.Errorf("cannot create node: %v", err)
+				}
+				// TODO: we could directly initialize the node with labels
+				if err := utils.AddLabelsToNode(cs, node.Name, map[string]string{"region": "test"}); err != nil {
+					return fmt.Errorf("cannot add labels to node: %v", err)
+				}
+				if _, err := createPausePod(cs, initPausePod(&pausePodConfig{Name: "pod-to-be-updated", Namespace: ns})); err != nil {
+					return fmt.Errorf("cannot create pod: %v", err)
+				}
+				return nil
+			},
+			pod: &pausePodConfig{
+				Name: "pod-1",
+				Affinity: &v1.Affinity{
+					PodAffinity: &v1.PodAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"pod-with-affinity": "true",
+									},
+								},
+								TopologyKey: "region",
+							},
+						},
+					},
+				},
+			},
+			update: func(cs kubernetes.Interface, ns string) error {
+				pod, err := getPod(cs, "pod-to-be-updated", ns)
+				if err != nil {
+					return fmt.Errorf("cannot get pod: %v", err)
+				}
+				pod.Labels = map[string]string{"pod-with-affinity": "true"}
+				if _, err := cs.CoreV1().Pods(pod.Namespace).Update(context.TODO(), pod, metav1.UpdateOptions{}); err != nil {
```

Comment:
@Huang-Wei @alculquicondor I have added this test as discussed above.

I could not find an easy way to update the pod. So I decided to use `getPod` and update the labels. An alternative would be to use `PodWrapper`, but it felt a bit weird to repeat all the pod fields again.

WDYT?

Reply:
Fetching it from API server looks good to me.